### DISCUSSION TITLE
Copying zip failed because: There is no disk set for the backup destination.

### DISCUSSION THREAD

**Discussion by DrGarciaDev** [spatie/laravel-backup#1105] [Q&A]
I got this exception when try to create backup using php artisan backup:run --only-db

```
C:\wamp64\www\crud-laravel-5.4>php artisan backup:run --only-db
Starting backup...
Dumping database crud_laravel_5.4...
Determining files to backup...
Zipping 1 files...
Created zip containing 1 files. Size is 22.12 KB
Copying zip to disk named local...
Successfully copied zip to disk named local.
Copying zip to disk named google...
Copying zip failed because: There is no disk set for the backup destination.
Backup completed!
```

My filesystems.php

```

<?php

return [

    /*
    |--------------------------------------------------------------------------
    | Default Filesystem Disk
    |--------------------------------------------------------------------------
    |
    | Here you may specify the default filesystem disk that should be used
    | by the framework. The "local" disk, as well as a variety of cloud
    | based disks are available to your application. Just store away!
    |
    */

    'default' => 'local',

    /*
    |--------------------------------------------------------------------------
    | Default Cloud Filesystem Disk
    |--------------------------------------------------------------------------
    |
    | Many applications store files both locally and in the cloud. For this
    | reason, you may specify a default "cloud" driver here. This driver
    | will be bound as the Cloud disk implementation in the container.
    |
    */

    'cloud' => 's3',

    /*
    |--------------------------------------------------------------------------
    | Filesystem Disks
    |--------------------------------------------------------------------------
    |
    | Here you may configure as many filesystem "disks" as you wish, and you
    | may even configure multiple disks of the same driver. Defaults have
    | been setup for each driver as an example of the required options.
    |
    | Supported Drivers: "local", "ftp", "s3", "rackspace"
    |
    */

    'disks' => [

        'local' => [
            'driver' => 'local',
            'root' => storage_path('app'),
        ],

        'public' => [
            'driver' => 'local',
            'root' => storage_path('app/public'),
            'url' => env('APP_URL').'/storage',
            'visibility' => 'public',
        ],

        's3' => [
            'driver' => 's3',
            'key' => env('AWS_KEY'),
            'secret' => env('AWS_SECRET'),
            'region' => env('AWS_REGION'),
            'bucket' => env('AWS_BUCKET'),
        ],

        'google' => [
            'driver' => 'google',
            'clientId' => env('GOOGLE_DRIVE_CLIENT_ID'),
            'clientSecret' => env('GOOGLE_DRIVE_CLIENT_SECRET'),
            'refreshToken' => env('GOOGLE_DRIVE_REFRESH_TOKEN'),
            'folderId' => env('GOOGLE_DRIVE_FOLDER_ID'),
        ],

    ],

];

```
 
a snippet laravel-bakup.php file
```

        'destination' => [

            /*
             * The filename prefix used for the backup zip file.
             */
            'filename_prefix' => '',

            /*
             * The disk names on which the backups will be stored.
             */
            'disks' => [
                'local',
                'google',
            ],
        ],
```

a snippet config/app.php 
```

        /*
        * Extern providers
        */
        Collective\Html\HtmlServiceProvider::class,
        Spatie\Backup\BackupServiceProvider::class,
        App\Providers\GoogleDriveServiceProvider::class,

```
follow this tutorial: 

[https://medium.com/@dennissmink/laravel-backup-database-to-your-google-drive-f4728a2b74bd](https://medium.com/@dennissmink/laravel-backup-database-to-your-google-drive-f4728a2b74bd)

I receive email notifications with this:

```



Whoops!

Importante: Ocurrió un error al realizar la copia de seguridad de Laravel

Mensaje de la excepción: There is no disk set for the backup destination
```
Traza de la excepción: #0 C:\wamp64\www\crud-laravel-5.4\vendor\spatie\laravel-backup\src\BackupDestination\BackupDestination.php(78): Spatie\Backup\Exceptions\InvalidBackupDestination::diskNotSet() #1 C:\wamp64\www\crud-laravel-5.4\vendor\spatie\laravel-backup\src\Tasks\Backup\BackupJob.php(239): Spatie\Backup\BackupDestination\BackupDestination->write('C:\wamp64\www\c...') #2 C:\wamp64\www\crud-laravel-5.4\vendor\laravel\framework\src\Illuminate\Support\Collection.php(287): Spatie\Backup\Tasks\Backup\BackupJob->Spatie\Backup\Tasks\Backup{closure}(Object(Spatie\Backup\BackupDestination\BackupDestination), 1) #3 C:\wamp64\www\crud-laravel-5.4\vendor\spatie\laravel-backup\src\Tasks\Backup\BackupJob.php(249): Illuminate\Support\Collection->each(Object(Closure)) #4 C:\wamp64\www\crud-laravel-5.4\vendor\spatie\laravel-backup\src\Tasks\Backup\BackupJob.php(137): Spatie\Backup\Tasks\Backup\BackupJob->copyToBackupDestinations('C:\wamp64\www\c...') #5 C:\wamp64\www\crud-laravel-5.4\vendor\spatie\laravel-backup\src\Commands\BackupCommand.php(49): Spatie\Backup\Tasks\Backup\BackupJob->run() #6 [internal function]: Spatie\Backup\Commands\BackupCommand->handle() #7 C:\wamp64\www\crud-laravel-5.4\vendor\laravel\framework\src\Illuminate\Container\BoundMethod.php(29): call_user_func_array(Array, Array) #8 C:\wamp64\www\crud-laravel-5.4\vendor\laravel\framework\src\Illuminate\Container\BoundMethod.php(87): Illuminate\Container\BoundMethod::Illuminate\Container{closure}() #9 C:\wamp64\www\crud-laravel-5.4\vendor\laravel\framework\src\Illuminate\Container\BoundMethod.php(31): Illuminate\Container\BoundMethod::callBoundMethod(Object(Illuminate\Foundation\Application), Array, Object(Closure)) #10 C:\wamp64\www\crud-laravel-5.4\vendor\laravel\framework\src\Illuminate\Container\Container.php(539): Illuminate\Container\BoundMethod::call(Object(Illuminate\Foundation\Application), Array, Array, NULL) #11 C:\wamp64\www\crud-laravel-5.4\vendor\laravel\framework\src\Illuminate\Console\Command.php(182): Illuminate\Container\Container->call(Array) #12 C:\wamp64\www\crud-laravel-5.4\vendor\symfony\console\Command\Command.php(255): Illuminate\Console\Command->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Illuminate\Console\OutputStyle)) #13 C:\wamp64\www\crud-laravel-5.4\vendor\laravel\framework\src\Illuminate\Console\Command.php(167): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Illuminate\Console\OutputStyle)) #14 C:\wamp64\www\crud-laravel-5.4\vendor\spatie\laravel-backup\src\Commands\BaseCommand.php(22): Illuminate\Console\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput)) #15 C:\wamp64\www\crud-laravel-5.4\vendor\symfony\console\Application.php(987): Spatie\Backup\Commands\BaseCommand->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput)) #16 C:\wamp64\www\crud-laravel-5.4\vendor\symfony\console\Application.php(255): Symfony\Component\Console\Application->doRunCommand(Object(Spatie\Backup\Commands\BackupCommand), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput)) #17 C:\wamp64\www\crud-laravel-5.4\vendor\symfony\console\Application.php(148): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput)) #18 C:\wamp64\www\crud-laravel-5.4\vendor\laravel\framework\src\Illuminate\Foundation\Console\Kernel.php(122): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput)) #19 C:\wamp64\www\crud-laravel-5.4\artisan(35): Illuminate\Foundation\Console\Kernel->handle(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput)) #20 {main}
```

**Comment by bbredewold** [comment]
Edit: My bad... was missing and environment variable...

~~Yips, I also have this. Stangest thing is, that when I run the command locally, it works fine. When I run the command through a cronjob, I get the error... I have 2 disks, backup (local) and backup-cloud (azure storage).~~

I'm using `5.12.1`

```
Running scheduled command: '/usr/bin/php7.2' 'artisan' backup:run >> '/root/test.log' 2>&1

Starting backup...
Dumping database projectwebsite...
Determining files to backup...
Zipping 52 files...
Created zip containing 52 files. Size is 2.44 MB
Copying zip to disk named backup...
Successfully copied zip to disk named backup.
Copying zip to disk named backup-cloud...
Copying zip failed because: There is no disk set for the backup destination.
Backup completed!
```
